test/webrick/test_https.rb

```diff
@@ -109,4 +109,89 @@ def test_check_ssl_virtual
       end
     }
   end
+
+  def test_ssl_meta_vars
+    # CA cert
+    ca_cert, ca_key = WEBrick::Utils.create_self_signed_cert(2048, "/CN=ca", "is CA")
+    ef = OpenSSL::X509::ExtensionFactory.new
+    ef.subject_certificate = ca_cert
+    ef.issuer_certificate = ca_cert
+    ca_cert.extensions = [
+      ef.create_extension("basicConstraints", "CA:TRUE", true),
+      ef.create_extension("keyUsage", "keyCertSign, cRLSign", true),
+      ef.create_extension("subjectKeyIdentifier", "hash", false)
+    ]
+    ca_cert.add_extension ef.create_extension("authorityKeyIdentifier", "keyid:always,issuer:always")
+    ca_cert.sign(ca_key, "SHA256")
+
+    # Client cert
+    client_cert, client_key = WEBrick::Utils.create_self_signed_cert(2048, "/CN=client", "is client")
+    client_cert.issuer = ca_cert.issuer
+    ef = OpenSSL::X509::ExtensionFactory.new
+    ef.subject_certificate = client_cert
+    ef.issuer_certificate = ca_cert
+    client_cert.extensions = [
+      ef.create_extension("basicConstraints", "CA:FALSE", true),
+      ef.create_extension("keyUsage", "digitalSignature", true),
+      ef.create_extension("subjectKeyIdentifier", "hash", false),
+      ef.create_extension("subjectAltName", "DNS:localhost,IP:127.0.0.1", false)
+    ]
+    client_cert.sign(ca_key, "SHA256")
+
+
+    # Server cert
+    server_cert, server_key = WEBrick::Utils.create_self_signed_cert(2048, "/CN=server", "is server")
+    server_cert.issuer = ca_cert.issuer
+    ef = OpenSSL::X509::ExtensionFactory.new
+    ef.subject_certificate = server_cert
+    ef.issuer_certificate = ca_cert
+    server_cert.extensions = [
+      ef.create_extension("basicConstraints", "CA:FALSE", true),
+      ef.create_extension("keyUsage", "digitalSignature", true),
+      ef.create_extension("subjectKeyIdentifier", "hash", false),
+      ef.create_extension("subjectAltName", "DNS:localhost,IP:127.0.0.1", false)
+    ]
+    server_cert.sign(ca_key, "SHA256")
+
+    # Client CA Store
+    ca_client_store = OpenSSL::X509::Store.new
+    ca_client_store.add_cert(ca_cert)
+    ca_client_store.add_cert(client_cert)
+
+    # Server CA Store
+    server_ca_store = OpenSSL::X509::Store.new
+    server_ca_store.add_cert(ca_cert)
+    server_ca_store.add_cert(server_cert)
+
+    config = {
+      SSLEnable: true,
+      :SSLCertName => "/CN=localhost",
```

Comment:
Can this key also used the non-hashrocket style? Or is it somehow special?

Reply:
nothing special about it, I just copy-pasted as is from the test above. I assumed that consistency wasn't a big issue in this codebase. Perhaps you should consider using rubocop at some point?